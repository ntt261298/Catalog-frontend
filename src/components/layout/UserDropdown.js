import React, { Component, Fragment } from 'react';
import { UncontrolledDropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';

class UserDropdown extends Component {
    render() {
        return (
            <Fragment>
                <UncontrolledDropdown>
                  <DropdownToggle caret>
                    <div className="user-img">
                      <img src="assets/images/baseline-person-24px.svg" />
                    </div>
                  </DropdownToggle>
                  <DropdownMenu>
                    {/* <Link>    */}
                      <DropdownItem>
                        Activities
                      </DropdownItem>
                    {/* </Link> */}
                    <DropdownItem divider />
                    <DropdownItem>Logout</DropdownItem>
                  </DropdownMenu>
                </UncontrolledDropdown>
            </Fragment>
        )
    }
}

UserDropdown.propTypes = {
    // toggleLogin: PropTypes.func.isRequired,
    // userLogout: PropTypes.func.isRequired,
    // getSearchResults: PropTypes.func.isRequired
  }
  
  const mapStateToProps = state => ({

  })
  export default connect(null, null)(UserDropdown);