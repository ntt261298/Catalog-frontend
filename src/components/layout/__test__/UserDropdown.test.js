import React from 'react';
import { shallow } from '../../../enzyme';
import { UserDropdown } from '../UserDropdown';

describe('src/coponents/layout/UserDropdown.js', () => {
  let wrapper;
  let props;
  let logoutLink;

  const update = () => {
    logoutLink = wrapper.find('.logout');
  };
  const setup = () => {
    wrapper = shallow(
      <UserDropdown {...props} />,
    );
    update();
  };

  beforeEach(() => {
    props = {
      signupModal: false,
      onLogout: jest.fn(),
    };
  });

  it('should render user dropdown', () => {
    setup();
    // Expect the wrapper object to be defined
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onLogout when logoutLink is clicked', () => {
    setup();
    logoutLink.simulate('click');
    // Expect the wrapper object to be defined
    expect(props.onLogout).toMatchSnapshot();
  });
});
