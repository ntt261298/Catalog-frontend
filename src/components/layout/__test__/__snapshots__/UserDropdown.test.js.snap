// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/coponents/layout/UserDropdown.js should call onLogout when logoutLink is clicked 1`] = `[Function]`;

exports[`src/coponents/layout/UserDropdown.js should render user dropdown 1`] = `
<UncontrolledDropdown>
  <DropdownToggle
    aria-haspopup={true}
    caret={true}
    color="secondary"
  >
    <div
      className="user-img"
    >
      <img
        alt="user"
        src="baseline-person-24px.svg"
      />
    </div>
  </DropdownToggle>
  <DropdownMenu
    flip={true}
    tag="div"
  >
    <Link
      to="/me/items"
    >
      <DropdownItem
        tag="button"
        toggle={true}
      >
        Activities
      </DropdownItem>
    </Link>
    <DropdownItem
      divider={true}
      tag="button"
      toggle={true}
    />
    <DropdownItem
      className="logout"
      onClick={[Function]}
      tag="button"
      toggle={true}
    >
      Logout
    </DropdownItem>
  </DropdownMenu>
</UncontrolledDropdown>
`;
